exchanges/bybit/bybit_ws_cfutures.go

```diff
@@ -0,0 +1,802 @@
+func (by *Bybit) WsCoinAuth() error {
+	intNonce := (time.Now().Unix() + 1) * 1000
+	strNonce := strconv.FormatInt(intNonce, 10)
+	hmac, err := crypto.GetHMAC(
+		crypto.HashSHA256,
+		[]byte("GET/realtime"+strNonce),
+		[]byte(by.API.Credentials.Secret),
+	)
+	if err != nil {
+		return err
+	}
+	sign := crypto.HexEncodeToString(hmac)
+	req := Authenticate{
+		Operation: "auth",
+		Args:      []string{by.API.Credentials.Key, strNonce, sign},
+	}
+	return by.Websocket.Conn.SendJSONMessage(req)
+}
+
+// SubscribeCoin sends a websocket message to receive data from the channel
+func (by *Bybit) SubscribeCoin(channelsToSubscribe []stream.ChannelSubscription) error {
+	var errs common.Errors
+	for i := range channelsToSubscribe {
+		var sub WsFuturesReq
+		sub.Topic = subscribe
+
+		sub.Args = append(sub.Args, formatArgs(channelsToSubscribe[i].Channel, channelsToSubscribe[i].Params))
+		err := by.Websocket.Conn.SendJSONMessage(sub)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		by.Websocket.AddSuccessfulSubscriptions(channelsToSubscribe[i])
+	}
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+func formatArgs(channel string, params map[string]interface{}) string {
+	argStr := channel
+	for x := range params {
+		argStr += dot + fmt.Sprintf("%v", params[x])
+	}
+	return argStr
+}
+
+// UnsubscribeCoin sends a websocket message to stop receiving data from the channel
+func (by *Bybit) UnsubscribeCoin(channelsToUnsubscribe []stream.ChannelSubscription) error {
+	var errs common.Errors
+
+	for i := range channelsToUnsubscribe {
+		var unSub WsFuturesReq
+		unSub.Topic = unsubscribe
+
+		a, err := by.GetPairAssetType(channelsToUnsubscribe[i].Currency)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		formattedPair, err := by.FormatExchangeCurrency(channelsToUnsubscribe[i].Currency, a)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		unSub.Args = append(unSub.Args, channelsToUnsubscribe[i].Channel+dot+formattedPair.String())
+		err = by.Websocket.Conn.SendJSONMessage(unSub)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		by.Websocket.RemoveSuccessfulUnsubscriptions(channelsToUnsubscribe[i])
+	}
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+// wsCoinReadData gets and passes on websocket messages for processing
+func (by *Bybit) wsCoinReadData() {
+	by.Websocket.Wg.Add(1)
+	defer by.Websocket.Wg.Done()
+
+	for {
+		select {
+		case <-by.Websocket.ShutdownC:
+			return
+		default:
+			resp := by.Websocket.Conn.ReadMessage()
+			if resp.Raw == nil {
+				return
+			}
+
+			err := by.wsCoinHandleData(resp.Raw)
+			if err != nil {
+				by.Websocket.DataHandler <- err
+			}
+		}
+	}
+}
+
+func (by *Bybit) wsCoinHandleData(respRaw []byte) error {
+	var multiStreamData map[string]interface{}
+	err := json.Unmarshal(respRaw, &multiStreamData)
+	if err != nil {
+		return err
+	}
+
+	if t, ok := multiStreamData["topic"].(string); ok {
```

Comment:
Can we invert this check?

```go
	t, ok := multiStreamData["topic"].(string)
	if ok {
		return errors.New("no topics found")
	}
```

Reply:
`
	t, ok := multiStreamData["topic"].(string)
	if !ok {
		return errors.New("no topics found")
	}
`

I think you meant this. Right?

Reply:
@Jdpurohit Yes, correct.